pkg/migrations/mls/20240829001344_serial-ids.up.sql

```diff
@@ -0,0 +1,44 @@
+CREATE FUNCTION insert_group_message(group_id BYTEA, data BYTEA, group_id_data_hash BYTEA)
+	RETURNS SETOF group_messages
+	AS $$
+BEGIN
+	-- Ensures that the generated sequence ID matches the insertion order
+	-- Only released at the end of the enclosing transaction - beware if called within a long transaction
+	PERFORM
+		pg_advisory_xact_lock(hashtext('group_messages_sequence'), hashtext(encode(group_id, 'hex')));
```

Comment:
I can't find documentation on how `pg_advisory_xact_lock` works when you provide two keys. Does it lock on both, or the combination of the two?